### PR TITLE
Feature/working example

### PR DESCRIPTION
**Pull request by maciejburda**
Added working example that's using current SDK version.

### CODE REVIEW

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 1]
```diff
@@ -10,7 +10,15 @@ import Foundation
 import CoreLocation
 import Alps
 
-open class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
+public typealias OnMatchClojure = (_ matches: [Match], _ device: Device) -> Void
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift]: Let's change to `closure`

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 2]
```diff
@@ -10,7 +10,15 @@ import Foundation
...
+
+public protocol AlpsManagerDelegate: class {
+    var onMatch: OnMatchClojure { get set }
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift]: `closure`

File: AlpsSDK/AlpsManager/AlpsManager.swift [hunk 3]
```diff
@@ -20,23 +28,22 @@ open class AlpsManager: MatchMonitorDelegate, ContextManagerDelegate {
...
-    lazy var publications = PublicationRepository()
-    lazy var subscriptions = SubscriptionRepository()
+    public lazy var publications = PublicationRepository()
+    public lazy var subscriptions = SubscriptionRepository()
     
-    lazy var locationUpdateManager = LocationUpdateManager()
+    public lazy var locationUpdateManager = LocationUpdateManager()
 
-    init(apiKey: String, baseUrl: String? = nil) {
+    public init(apiKey: String, baseURL: String? = nil) {
```
Comment (wenjdu) [inline AlpsSDK/AlpsManager/AlpsManager.swift:42]: I also made public almost all the members and many class, when using SDK to try Proximity Handler.
We can keep them all open for the moment and we will look for private when we talk about security

File: AlpsSDK/ModelExtensions.swift [hunk 4]
```diff
@@ -53,11 +52,9 @@ extension Device: Hashable {
     }
 }
 
-extension Publication {
-
-    internal convenience init(deviceId: String? = nil, topic: String, range: Double, duration: Double, properties: [String: String]) {
+public extension Publication {
+    public convenience init(deviceId: String? = nil, topic: String, range: Double, duration: Double, properties: [String: String]) {
```
Comment (wenjdu) [inline AlpsSDK/ModelExtensions.swift:56]: If there is no deviceId, it shouldn't be able to init

File: README.md [hunk 5]
```diff
@@ -4,16 +4,16 @@
 
 ## Versioning
 
-Only compatible with Swift 3.
+SDK is written using Swift 4.
 
-Alps SDK requires iOS 7+, if you want to use iBeacons.
+Alps SDK requires iOS 9+.
```
Comment (wenjdu) [inline README.md]: 10 +

File: README.md [hunk 6]
```diff
@@ -31,101 +31,59 @@ Everytime you call an asynchronous function and it succeeds, our SDK stores it.
...
+            print("We're all good! 🏔")
+        })
+    })                    
 }
 ```
 
-Here is an example on how to create your first user with device and get a match.
-Please refer to documentation "tutorial" to get a full explanation on this example.
-
+Create a AlpsManagerDelegate that has `OnMatchClojure`.
```
Comment (wenjdu) [inline README.md]: `closure`

File: README.md [hunk 7]
```diff
@@ -31,101 +31,59 @@ Everytime you call an asynchronous function and it succeeds, our SDK stores it.
...
-
-// Create a subscription
-let selector = "mood = 'happy'"
-self.appDelegate.alps.createSubscription(topic: "tutorial", selector: selector, range: 1000, duration: 300, completion: {
-(_ subscription) in
-if let s = subscription {
-print(s.id)
+class MatchDelegate: AlpsManagerDelegate {
+    var onMatch: OnMatchClojure
+    init(_ onMatch: @escaping OnMatchClojure) {
```
Comment (wenjdu) [inline README.md]: same

**Review by wenjdu — APPROVED** [review]

File: AlpsSDK/ModelExtensions.swift (hunk 4 above)
Comment (maciejburda) [reply AlpsSDK/ModelExtensions.swift:56]: It's not entirely clear, but if you have a `mainDevice` it's id will be added to publication.

File: AlpsSDK/AlpsManager/AlpsManager.swift (hunk 3 above)
Comment (maciejburda) [reply AlpsSDK/AlpsManager/AlpsManager.swift:42]: Agreed.

**Pull request merged**